week2/The system of the student's score management. v1.0.cpp

```diff
@@ -0,0 +1,172 @@
+#include <stdio.h>
+
+void readscore(long student_ID[], int score[], int n);
+void print_total_and_average(int n, int score[]);
+void rank_by_grade_in_descending_order(long student_ID[], int score[], int n);
+void print_score(long student_ID[], int score[], int n);
+void rank_by_student_ID_in_ascending_order(long student_ID[], int score[], int n);
+int search_student_rank_and_grade_by_student_ID(long student_ID[], int score[], int n);
+void count_the_percentage_of_the_each_category(long student_ID[], int score[], int n);
+
+int main()
+{
+	long student_ID[30];
+	int score[30], n, choice;
+	printf("1.Input record\n");
+	printf("2.Caculate total and average score of course\n");
+	printf("3.Sort in descending order by score\n");
+	printf("4.Sort in ascending order by number\n");
+	printf("5.search by number\n");
+	printf("6.Statistic analysis\n");
+	printf("0.Exit\n\n");
+	do {
+		printf("Please enter your choice:");
+		scanf("%d", &choice);
+		switch (choice)
+		{
+			case 1:
+				printf("Please input the number of the students:");
+				scanf("%d", &n);
+				if (n > 30) {
+					printf("The number is beyond\n");

```

Comment:
This message is not good to program user. How do the user know about the allowed range? Besides, `beyond` is not a good expression, use exceed limit instead.

Principal here is always designing and implementing your program from users' perspective